add module name option for generator commands to support #855

munza opened the pull request:
add —module (-m) option for administrate generators that will create generated file under the specified module name. eg. rails g administrate:install -m Manage.

This will solve the problem if anyone already has a model named `Admin` in their application and wish to use a different module name other than the default `Admin` module of the Administrate.

We can now use the `--module` (`-m` as shorthand) and then specify the module name we wish to use in the generator command like below -
- `rails g administrate:install -m Manage`
- `rails g administrate:dashboard Foo -m Manage`
etc.

We can then see that the generated files are using the specified `Manage` module instead of the default `Admin` module namespacing.

nickcharlton commented:
Thanks for this!

1. Could I get you to fix up those style comments?
2. What do you think about using the example "Manage" instead of "Backend"?
    - I feel like "Backend" might be more of a broader term for the web side of a project, and "Manage" might be a bit clearer?

munza commented:
@nickcharlton thanks for your feedback. I did not like the "Backend" word either. But I could not find any suitable word other than that at that time. I have changed the example into "Manage" from "Backend" and also in the spec tests because it sounds better.
and I have fixed the style issues.

But CI test is failing where it is passing (`rake spec`) in my local machine. Although I needed to add `host`, `port` and `username` in the `spec/example_app/config/database.yml` file. Which I have removed (back to original state) at the last commit. What am I doing wrong that is failing the tests at CI?

tysongach commented on README.md:
````diff
@@ -55,18 +55,26 @@ Re-bundle, then run the installer:
 ```bash
 $ rails generate administrate:install
 ```
+Or to use a custom module name (default is `Admin`). eg. `Backend`
````
Add one empty line above.

tysongach commented on README.md:
````diff
@@ -55,18 +55,26 @@ Re-bundle, then run the installer:
 $ rails generate administrate:install
 ```
+Or to use a custom module name (default is `Admin`). eg. `Backend`
+```bash
````
Add one empty line above.

tysongach commented on README.md:
````diff
@@ -55,18 +55,26 @@ Re-bundle, then run the installer:
 ```bash
 $ rails generate administrate:dashboard Foo
 ```
+Or to use a custom module name (default is `Admin`). eg. `Backend`
````
Add one empty line above.

tysongach commented on README.md:
````diff
@@ -55,18 +55,26 @@ Re-bundle, then run the installer:
 $ rails generate administrate:dashboard Foo
 ```
+Or to use a custom module name (default is `Admin`). eg. `Backend`
+```bash
````
Add one empty line above.

nickcharlton commented:
@munza I'm seeing failures with:

```sh
rspec ./spec/generators/dashboard_generator_spec.rb:438 # Administrate::Generators::DashboardGenerator resource controller subclasses Admin::ApplicationController
rspec ./spec/generators/dashboard_generator_spec.rb:429 # Administrate::Generators::DashboardGenerator resource controller has valid syntax
rspec ./spec/generators/dashboard_generator_spec.rb:454 # Administrate::Generators::DashboardGenerator resource controller accepts module name as options
```

when running locally. Could I get you to look into those?

munza commented:
Thanks, @nickcharlton, I have fixed those 3 failures but the ci/circleci has reported more failures which are not appearing in my local build. Those failures are all related to file read issues. Each one of them has the same issue with different file location. Take a look at of the fails for an example -
```
Administrate::Generators::Views::FormGenerator administrate:views:form resource copies the form view into the `admin/resource` namespace
     Failure/Error: contents = File.read(file("app/views/admin/users/_form.html.erb"))
     
     Errno::ENOENT:
       No such file or directory @ rb_sysopen - /home/ubuntu/administrate/tmp/app/views/admin/users/_form.html.erb
     # ./spec/generators/views/form_generator_spec.rb:22:in `read'
     # ./spec/generators/views/form_generator_spec.rb:22:in `block (3 levels) in <top (required)>'
```
I am totally lost on these issues on how to fix these? Is it related with the file permission? I didn't change anything related to permission.

munza commented:
@tysongach, sorry I forgot to add your markdown changes. But I will add them in the next push. Thanks for the suggestion.

nickcharlton commented:
I just ran this locally, and I also can't see the errors we've got on CI.

We've got some failures which we've fixed on `master`, so I'd recommend rebasing against that and then pushing up again. The CI errors should go with it.

munza commented:
I have fetched the master branch from the original repo but the previous errors followed me there too. Looks like I am out of luck.
I will try to find a solution for this but others' help will be highly appreciated.

nickcharlton commented:
@munza It's made a little more complex as your commits are on `master`, but some git dancing can fix that up. I've gone ahead and done that (You have [enable contributor pushes](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) enabled). So we'll see if this passes.

nickcharlton commented:
Huh! So we're still seeing those errors on CI. I'm going to reach out to Circle and see what's up.

nickcharlton commented:
I'm going to close this as we merged #956 which achieves the same thing as this PR and we never got to the bottom of those file issues.
